gcp-observability/src/test/java/io/grpc/gcp/observability/TracesTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.gcp.observability;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.trace.v1.TraceServiceClient;
+import com.google.cloud.trace.v1.TraceServiceClient.ListTracesPagedResponse;
+import com.google.devtools.cloudtrace.v1.GetTraceRequest;
+import com.google.devtools.cloudtrace.v1.ListTracesRequest;
+import com.google.devtools.cloudtrace.v1.Trace;
+import com.google.devtools.cloudtrace.v1.TraceSpan;
+import com.google.protobuf.util.Timestamps;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.gcp.observability.interceptors.InternalLoggingChannelInterceptor;
+import io.grpc.gcp.observability.interceptors.InternalLoggingServerInterceptor;
+import io.grpc.gcp.observability.logging.GcpLogSink;
+import io.grpc.gcp.observability.logging.Sink;
+import io.grpc.testing.GrpcCleanupRule;
+import io.grpc.testing.protobuf.SimpleServiceGrpc;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class TracesTest {
+
+  private static final String PROJECT_ID = "PROJECT";
+  private String customTagKey = "service";
+  private String customTagValue =
+      String.format("payment-%s", String.valueOf(System.currentTimeMillis()));
+  private Map<String, String> customTags = Collections.singletonMap(customTagKey, customTagValue);
+
+  @Rule public final GrpcCleanupRule cleanupRule = new GrpcCleanupRule();
+
+  /**
+   * Cloud Trace test using GlobalInterceptors.
+   *
+   * <p>Ignoring test, because it calls external Cloud Tracing APIs. To test cloud trace setup
+   * locally, 1. Set up Cloud auth credentials 2. Assign permissions to service account to write
+   * traces to project specified by variable PROJECT_ID 3. Comment @Ignore annotation
+   */
```

Comment:
Similar to the other comments, it might be worth mentioning that the test has been verified manually

Reply:
Added. PTAL